### ISSUE TITLE
any plan on adding local attention?

### ISSUE THREAD

**Issue by NaxAlpha** [lucidrains/gated-state-spaces-pytorch#3]
Hi,

I have been experimenting with the model. It sounds very promising given it does not have similar limitation as of attention based models. I have written [a blog post](https://naxalpha.substack.com/p/devlog-experiment-a2a468-gated-state) on experiments with this model as well as made [the pretrained model](https://huggingface.co/naxalpha/gated-state-space) available on HuggingFace. Since the training is still going on, [a report](https://wandb.ai/naxalpha/gated-state-space/reports/Gated-State-Space-Training-v1--VmlldzozMTYzMzY3?accessToken=zy10rrpofi9k7l52aqwiej8bk0ub302rdswfkxmf8y94dt2j6z4kxbca6ar3sc52) is also available on W&B with the on going loss curve and generated samples. I am interested in learning how much attention improves the convergence.
